test/testdrive/source-linear-operators.td

```diff
@@ -0,0 +1,38 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test
+
+$ set schema={
+    "type": "record",
+    "name": "row",
+    "fields": [
+      {"name": "a", "type": "long"},
+      {"name": "b", "type": "long"}
+    ]
+  }
+
+$ kafka-create-topic topic=data
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"a": 1, "b": 1}
+{"a": 2, "b": 1}
+{"a": 3, "b": 1}
+{"a": 1, "b": 2}
+
+> CREATE SOURCE data
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${schema}'
+
+> CREATE VIEW data_view as SELECT * from data
+
+$ set-regex match=u\d+ replacement=UID
+
+> EXPLAIN PLAN FOR SELECT * from data_view where a = 1
+"Source materialize.public.data (UID):\n| Filter (#0 = 1)\n| Project (#0, #1)\n\nQuery:\n%0 =\n| Get materialize.public.data (UID)\n| Filter (#0 = 1)\n"
```

Comment:
Oof, that is thoroughly impossible to read. Sad, but seems fine for now. I wonder if @philip-stoev has thoughts on making this better.